Source/WTF/wtf/MemoryPressureHandler.cpp

```diff
@@ -123,6 +123,11 @@ std::optional<size_t> MemoryPressureHandler::thresholdForMemoryKill()
     if (m_configuration.killThresholdFraction)
         return m_configuration.baseThreshold * (*m_configuration.killThresholdFraction);
 
+#if PLATFORM(GTK) || PLATFORM(WPE)
+    // We don't want to kill the process unless there's a killThreshold set in the configuration.
+    return std::nullopt;
+#endif
```

Comment:
Eh, if the application uses WebKitMemoryPressureSettings, then of course we need to respect that. But most applications won't ever configure that. My vote would be to have a good default and kill the runaway process anyway. So if the application sets a custom kill threshold, that's handled above, and if there isn't, we would use the default one. i.e. I wouldn't change anything here.

That leaves one confusing problem: the default kill threshold of WebKitMemoryPressureSettings is 0, which doesn't match the internals. That's confusing.